x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -259,17 +262,23 @@ static class ConvertStringToByteRef extends OptimizerRules.OptimizerExpressionRu
 
         @Override
         protected Expression rule(Literal lit) {
-            if (lit.value() == null) {
+            Object value = lit.value();
+
+            if (value == null) {
                 return lit;
             }
-            if (lit.value() instanceof String s) {
+            if (value instanceof String s) {
                 return Literal.of(lit, new BytesRef(s));
             }
-            if (lit.value() instanceof List<?> l) {
+            if (value instanceof List<?> l) {
                 if (l.isEmpty() || false == l.get(0) instanceof String) {
                     return lit;
                 }
-                return Literal.of(lit, l.stream().map(v -> new BytesRef((String) v)).toList());
+                List<BytesRef> byteRefs = new ArrayList<>(l.size());
+                for (Object v : l) {
+                    byteRefs.add(new BytesRef(v.toString()));
+                }
+                return Literal.of(lit, byteRefs);
             }
             return lit;
         }
```

Comment:
Unrelated change - couldn't help not correct it to save the method invocation and replace the noise map with a good-ol' reliable iteration.